### PR TITLE
Update index route to return server commit hash

### LINKED ISSUE #14: The index route should return metadata about the service itself
Currently the index route attempts to make an authenticated API call.

### PR DESCRIPTION
**Pull request by shenanigansd**
Closes #14

### CODE REVIEW

File: src/reporter/models.py [hunk 1]
```diff
@@ -6,6 +6,13 @@
 from pydantic import BaseModel, model_validator
 
 
+class ServerMetadata(BaseModel):
+    """Metadata about the server."""
```
Comment (jonathan-d-zhang) [inline src/reporter/models.py:10]:
```suggestion
    """Server metadata."""
```

File: src/reporter/app.py [hunk 2]
```diff
@@ -18,11 +19,13 @@
 app = FastAPI()
 
 
-@app.get("/")
-async def echo(http_client: HTTPClientDependency) -> str:
-    """Return the username of the PyPI User."""
-    response = await http_client.get("/echo")
-    return response.text
+@app.get("/", summary="Get the server's metadata")
```
Comment (jonathan-d-zhang) [inline src/reporter/app.py:22]:
```suggestion
@app.get("/", summary="Get server metadata")
```

**Review by jonathan-d-zhang — CHANGES_REQUESTED** [review]
I have never nitted harder in my life.

File: src/reporter/app.py [hunk 3]
```diff
@@ -18,11 +19,13 @@
...
+async def metadata() -> ServerMetadata:
+    """Get server metadata."""
+    return ServerMetadata(
+        version=__version__,
+        server_commit=GIT_SHA,
```
Comment (sid-maddy) [inline src/reporter/app.py:27]: Just `commit` works, right?

Comment (shenanigansd) [reply src/reporter/app.py]: Sure

**Review by jonathan-d-zhang — APPROVED** [review]
:shipit:

**Pull request merged**